src/OpenTelemetry/Internal/TagAndValueTransformer.cs

```diff
@@ -44,7 +47,7 @@ protected TValue TransformValue(object value)
                 return this.TransformFloatingPointValue(Convert.ToDouble(value));
             default:
                 // This could throw an exception. The caller is expected to handle.
-                return this.TransformStringValue(value.ToString());
+                return this.TransformStringValue(value?.ToString());
```

Comment:
There is the `Debug.Assert` on line 28 so it seems like the intention here `null` is not allowed and enforced elsewhere. What I would suggest in that case is not allow a `null` downstream. Just make it:

```diff
- return this.TransformStringValue(value?.ToString());
+ return this.TransformStringValue(value!.ToString()); // The ! is needed for older runtimes where Debug.Assert does not have null decoration. Like netstandard2.0 and net462.

- protected abstract TValue TransformStringValue(string? value);
+ protected abstract TValue TransformStringValue(string value);
```

Reply:
I thing that we have two topics here.
1. `value` detected as null. - I agree that it should not happen. Can we just remove Debug.Assert? The compiler should keep code safe.
2. Changing TransformStringValue contract. Technically. Object.ToString() is defined as `public virtual string? ToString()` (in .NET 6). What is your recommendation for handling such case?